File: OnlineBank.LoanService.Common/Dtos/Integrations/CreateTransactionMessage.cs
namespace OnlineBank.LoanService.Common.Dtos.Integrations;

public class CreateTransactionMessage
{
    public decimal amount { get; set; }
    public string transactionType { get; set; }
    public string currencyCode { get; set; }
    public Guid bankAccountId { get; set; }
    public Guid? fromBankAccountId { get; set; }
    public Guid? toBankAccountId { get; set; }
    public Guid authenticatedUserId { get; set; }
}

File: OnlineBank.CoreService/src/main/java/ru/hits/coreservice/dto/CreateTransactionMessage.java
package ru.hits.coreservice.dto;

import lombok.*;
import ru.hits.coreservice.enumeration.DepositTransactionType;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import java.math.BigDecimal;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class CreateTransactionMessage {

    private BigDecimal amount;

    private String transactionType;

    private UUID bankAccountId;

    private UUID fromBankAccountId;

    private UUID toBankAccountId;

    private String currencyCode;

}
